omero/developers/json-api.txt

```diff
@@ -0,0 +1,526 @@
+          "version": "0",
+          "url:base": "http://<server>/api/v0/"
+        }
+      ]
+    }
+
+
+List starting URLs
+^^^^^^^^^^^^^^^^^^
+
+The base URL for the chosen version will list a number of URLs for
+logging on and getting started.
+
+::
+
+    GET     /api/v0/
+
+
+**Response**
+
+::
+
+    {
+      "url:login": "http://<server>/api/v0/login/",
+      "url:save": "http://<server>/api/v0/m/save/",
+      "url:projects": "http://<server>/api/v0/m/projects/",
+      "url:plates": "http://<server>/api/v0/m/plates/",
+      "url:datasets": "http://<server>/api/v0/m/datasets/",
+      "url:token": "http://<server>/api/v0/token/",
+      "url:schema": "http://www.openmicroscopy.org/Schemas/OME/2016-06",
+      "url:screens": "http://<server>/api/v0/m/screens/",
+      "url:servers": "http://<server>/api/v0/servers/",
+      "url:images": "http://<server>/api/v0/m/images/"
+    }
+
+
+List available OMERO servers
+----------------------------
+
+Your API may allow you to connect to several different OMERO servers.
+
+::
+
+    GET     /api/v0/servers/
+
+
+**Response**
+
+::
+
+
+
+    {
+      "data": [
+        {
+          "host": "<server>",
+          "server": "omero",
+          "id": 1,
+          "port": 4064
+        }
+      ]
+    }
+
+
+Get CSRF token
+--------------
+
+In order to prevent CSRF attacks, CSRF tokens are required for any
+POST, PUT and DELETE requests. You will need to obtain a CSRF token
+for your session and use it for all subsequent requests in that session.
+You can obtain this from the ``csrftoken``
+cookie of any request or from the following URL:
+
+::
+
+    GET     /api/v0/token/
+
+
+**Response**
+
+::
+
+    {
+      "data": "eNoVq528bOqlhQqbCzKuviODTRX3PUO2"
+    }
+
+
+Login
+-----
+
+You can login to create an OMERO session. You must also include the CSRF token,
+either in the POST parameters as 'csrfmiddlewaretoken' or in the session
+header as 'X-CSRFToken'.
+
+The EventContext for this session will be returned to you.
+
+::
+
+    POST    /api/v0/login/
+
+
+**Parameters**
+
+::
+
+    Name                  Type        Description
+    ------------------------------------------------------------------
+    server                Number      ID of the server
+    username              String      User's username
+    password              String      User's password
+    csrfmiddlewaretoken   String      CSRF token (can be provied in header)
+
+
+**Response**
+
+::
+
+    {
+      "eventContext": {
+        "userName": "ben",
+        "eventId": -1,
+        "sessionUuid": "0b30ee4a-c0b2-4b0f-9c61-f48b31bcad8c",
+        "eventType": "User",
+        "userId": 3,
+        "sessionId": 171319,
+        "groupName": "Nevis Lab",
+        "isAdmin": False,
+        "memberOfGroups": [5, 1, 4],
+        "leaderOfGroups": [],
+        "groupId": 5
+      },
+      "success": true
+    }
+
+
+Image containers
+----------------
+
+OMERO organises images in two types of many-to-many hierarchies:
```

Comment:
maybe "hierarchy" as the "types of" is already plural? happy for @hflynn to decide that